ipsec/sha256.c

```diff
@@ -0,0 +1,470 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+
+/**
+  @file sha256.c
+  LTC_SHA256 by Tom St Denis
+*/
+#include "sha256.h"
+
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/minmax.h>
+
+
+#if defined(ENDIAN_BIG)
```

Comment:
We should patch this sucker up (along with #include <endian.h>)
```suggestion
#if __BYTE_ORDER == __BIG_ENDIAN
```

Reply:
I'm not patching it - looks like it's working. All crypto code will be probably replaced later.

Reply:
Yeah, it's working by accident - because all of our targets are litte-endian (for now) :P